util/grep.cwl

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env cwl-runner
+cwlVersion: v1.0
+class: CommandLineTool
+label: grep
+doc: print lines that match patterns
+$namespaces:
+  s: http://schema.org/
+
+requirements:
+  DockerRequirement:
+    dockerPull: busybox:latest
+  InlineJavascriptRequirement: {}
+
+inputs:
+  ignore_case:
+    doc: ignore case while matching
+    type: boolean?
+    inputBinding:
+      prefix: -i
+  inherit_format:
+    doc: copy format from search_target input to result output
+    type: boolean?
+  invert_selection:
+    doc: print non-matching lines
+    type: boolean?
+    inputBinding:
+      prefix: -v
+  line_no:
+    doc: add line number prefix to results
+    type: boolean?
+    inputBinding:
+      prefix: -n
+  lines_after:
+    doc: print up to N lines of trailing content
+    type: int?
+    inputBinding:
+      prefix: -A
+  lines_before:
+    doc: print up to N lines of leading content
+    type: int?
+    inputBinding:
+      prefix: -B
+  literal_pattern:
+    doc: interpret the pattern as a literal string, not a regular expression
+    type: boolean?
+    inputBinding:
+      prefix: -F
+  pattern:
+    doc: pattern to search for
+    type: string
+    inputBinding:
+      prefix: -e
+  search_target:
+    doc: file to search
+    type: File
+    inputBinding:
+      position: 100
+    streamable: true
+
+outputs:
+  result:
+    type: stdout
+    format: |-
+      $( inputs.inherit_format && inputs.search_target.format ? inputs.search_target.format : null )
+    streamable: true
+stdout: search_result$(inputs.search_target.nameext)
```

Comment:
Like with awk, I would leave the extension off; may not correlate with produced results.

Reply:
Fixed as per above not for `awk`.